src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java

```diff
@@ -32,4 +35,16 @@ public WalletDTO createWallet(CreateWalletDTO createWalletDTO) {
         throw new ResourceNotSavedException("Sorry. Something went wrong and your Wallet is not saved. Contact administrator.");
     }
 
+    @Override
+    public List<WalletDTO> getWallets() {
+        List<WalletDTO> walletsDTO;
+        try {
+            walletsDTO = walletRepository.findAll().stream()
+                    .map(walletModelMapper::mapWalletEntityToWalletDTO)
+                    .toList();
+        } catch (RuntimeException e) {
+            throw new ResourceNotFoundException("An error occurred while retrieving the list of wallets");
```

Comment:
I'd like to suggest you, to store exception messages in specified class (for example WalletExceptionsFinals) and then, this line could be more like this -> `throw new ResourceNotSavedException(EXAMPLE_MESSAGE);`

Reply:
Done, `WalletExceptionMessages` class has been added